driver-core/src/main/java/com/datastax/driver/core/Metrics.java

```diff
@@ -79,6 +79,16 @@ public Integer getValue() {
             return value;
         }
     });
+    private final Gauge<Integer> inFlightRequests = registry.register("inflight-requests", new Gauge<Integer>() {
+        @Override
+        public Integer getValue() {
+            int value = 0;
+            for (SessionManager session : manager.sessions)
+                for (HostConnectionPool pool : session.pools.values())
+                    value += pool.totalInFlight.get();
+            return value;
+        }
+    });
```

Comment:
lgtm.
My only remark is that the sum is not done atomically, but there would be no reasonable way to do it, and this should provide a good enough approximation.